sql/plan/insert.go

```diff
@@ -355,9 +378,58 @@ func (i insertIter) Close(ctx *sql.Context) error {
 			}
 		}
 	}
+
 	return nil
 }
 
+func (i *insertIter) updateLastInsertId(ctx *sql.Context, row sql.Row) {
+	if i.lastInsertIdUpdated {
+		return
+	}
+
+	var autoIncVal int64
+	var found bool
+	for i, expr := range i.insertExprs {
+		if _, ok := expr.(*expression.AutoIncrement); ok {
+			autoIncVal = toInt64(row[i])
+			found = true
+			break
+		}
+	}
+
+	if found {
+		ctx.SetLastQueryInfo(sql.LastInsertId, autoIncVal)
+		i.lastInsertIdUpdated = true
+	}
+}
+
+func toInt64(x interface{}) int64 {
+	switch x := x.(type) {
```

Comment:
Should add `int` and `uint` cases as well.

Reply:
Done